frontend/add-leads/add-edit-materials.component.tsx

```diff
@@ -0,0 +1,111 @@
+import { useCss } from "kremling";
+import React from "react";
+import css from "./add-edit-materials.css";
+import easyFetch from "../util/easy-fetch";
+import MaterialEntry from "./add-edit-materialEntry.component";
+
+export default function AddEditMaterials({
+  addMaterial,
+  deleteMaterial,
+  editMaterial,
+  materials,
+}) {
+  const [material, setMaterial] = React.useState({ name: "" });
+
+  function handleSubmit(evt) {
+    evt.preventDefault();
+
+    const abortController = new AbortController();
+    easyFetch(`/api/materials`, {
+      signal: abortController.signal,
+      method: "POST",
+      body: material,
+    })
+      .then((data) => {
+        addMaterial(data);
+        setMaterial({ name: "" });
+      })
+      .catch((err) => {
```

Comment:
The setTimeout pattern is now in a utility function:

```suggestion
      .catch(handlePromiseError)
```